### PR TITLE
Document and type PHP events

### PR DESCRIPTION
**Pull request by nickvergessen**
### ☑️ Resolves

* Base for https://github.com/nextcloud/spreed/pull/10729

### 🚧 Tasks

- [x] Migrate events to typed events
- [x] Type hint and document the parameters
- [x] Deprecate all other events
- [x] Add CI so we don't forget to document events again :-(

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not possible
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: lib/Config.php [hunk 1]
```diff
@@ -342,6 +343,8 @@ public function getTurnServers(bool $withEvent = true): array {
 		if ($withEvent) {
 			$event = new GetTurnServersEvent($servers);
 			$this->dispatcher->dispatchTyped($event);
+			$event = new BeforeTurnServersGetEvent($event->getServers());
+			$this->dispatcher->dispatchTyped($event);
```
Comment (come-nc) [inline lib/Config.php:347]: Are those not the same thing?

**Review by come-nc — APPROVED** [review]

File: lib/Config.php (hunk 1 above)
Comment (nickvergessen) [reply lib/Config.php:347]: `GetTurnServersEvent` is not following the suggested naming schema from https://docs.nextcloud.com/server/latest/developer_manual/basics/events.html#naming-scheme
So replaced it with a "valid" one :)

Comment (come-nc) [reply lib/Config.php:347]: Would it work to have one extend the other so we only dispatch once or does listener not work that way?

Comment (nickvergessen) [reply lib/Config.php:347]: the endpoint is only called once per visitor and I will drop the event once the internal listeners have been migrated, so I don't think it's worth any shenanigans

**Pull request merged**